sematic/plugins/abstract_metrics_storage.py

```diff
@@ -0,0 +1,202 @@
+"""
+Module declaring the interface for metrics storage plug-ins.
+"""
+# Standard Library
+import abc
+import enum
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import List, Literal, Optional, Sequence, Tuple, Type, Union, cast
+
+# Sematic
+from sematic.abstract_plugin import AbstractPlugin, PluginScope
+from sematic.config.settings import get_active_plugins
+from sematic.metrics.metric_point import MetricPoint, MetricsLabels
+
+
+@dataclass
+class MetricsFilter:
+    """
+    Data structure to query metrics.
+    """
+
+    name: str
+    from_time: datetime
+    to_time: datetime
+    labels: MetricsLabels
+
+
+@dataclass
+class MetricSeries:
+    """
+    Data structure to represent a metric series for queries to return.
+
+    Series are typically aggregated by labels.
```

Comment:
```suggestion
    Series can be aggregated by labels, and by `GroupBy` dimensions.
```

Reply:
GroupBy dimensions correspond to labels.